app/controllers/nonsipp/membercontributions/MemberContributionsCYAController.scala

```diff
@@ -95,41 +123,125 @@ class MemberContributionsCYAController @Inject()(
+        )
+      )
+    }
 }
 
-case class ViewModelParameters(
-  srn: Srn,
-  memberName: String,
-  index: Max300,
-  contributions: Money,
-  mode: Mode
-)
+//case class ViewModelParameters(
+//  srn: Srn,
+//  memberName: String,
+//  index: Max300,
+//  contributions: Money,
+//  mode: Mode,
+//  isPSP: Boolean,
+//  viewOnlyUpdated: Boolean,
+//  optYear: Option[String] = None,
+//  optCurrentVersion: Option[Int] = None,
+//  optPreviousVersion: Option[Int] = None,
+//  compilationOrSubmissionDate: Option[LocalDateTime] = None
+//)
 object MemberContributionsCYAController {
-  def viewModel(parameters: ViewModelParameters): FormPageViewModel[CheckYourAnswersViewModel] =
+  def viewModel(
+    srn: Srn,
+    memberName: String,
+    index: Max300,
+    contributions: Money,
+    mode: Mode,
+    isPSP: Boolean,
+    viewOnlyUpdated: Boolean,
+    optYear: Option[String] = None,
+    optCurrentVersion: Option[Int] = None,
+    optPreviousVersion: Option[Int] = None,
+    compilationOrSubmissionDate: Option[LocalDateTime] = None
+  ): FormPageViewModel[CheckYourAnswersViewModel] =
     FormPageViewModel[CheckYourAnswersViewModel](
-      title = parameters.mode
-        .fold(normal = "MemberContributionCYA.title", check = "MemberContributionCYA.change.title"),
-      heading = parameters.mode.fold(
+      mode = mode,
+      title = mode
+        .fold(
+          normal = "MemberContributionCYA.title",
+          check = "MemberContributionCYA.change.title",
+          viewOnly = "MemberContributionCYA.view.title"
+        ),
+      heading = mode.fold(
         normal = "MemberContributionCYA.heading",
         check = Message(
           "MemberContributionCYA.change.heading",
-          parameters.memberName
-        )
+          memberName
+        ),
+        viewOnly = "MemberContributionCYA.viewOnly.title"
       ),
       description = None,
       page = CheckYourAnswersViewModel(
         sections(
-          parameters.srn,
-          parameters.memberName,
-          parameters.index,
-          parameters.contributions,
-          CheckMode
+          srn,
+          memberName,
+          index,
+          contributions,
+          mode match {
+            case ViewOnlyMode => NormalMode
+            case _ => mode
+          }
         )
       ),
       refresh = None,
-      buttonText = parameters.mode.fold(normal = "site.saveAndContinue", check = "site.continue"),
+      buttonText =
+        mode.fold(normal = "site.saveAndContinue", check = "site.continue", viewOnly = "site.return.to.tasklist"),
       onSubmit = controllers.nonsipp.membercontributions.routes.MemberContributionsCYAController
-        .onSubmit(parameters.srn, parameters.index, parameters.mode)
+        .onSubmit(srn, index, mode),
+      optViewOnlyDetails = if (mode == ViewOnlyMode) {
+        Some(
+          ViewOnlyDetailsViewModel(
+            updated = viewOnlyUpdated,
+            link = (optYear, optCurrentVersion, optPreviousVersion) match {
```

Comment:
There is no previous link for the so called Tier 2 view only pages. IMO this can be just hardcoded to `link = None`. If that's the case, it might make the previous parameter (and the URL parameter `:previous` unnecessary.